keras/engine/training.py

```diff
@@ -68,6 +71,9 @@ def _standardize_input_data(data, names, shapes=None,
                                  str(names))
             arrays.append(data[name])
     elif isinstance(data, list):
+        for key, value in enumerate(data):
+            if value.__class__.__name__ == 'DataFrame':
+                data[key] = value.values
```

Comment:
When using `train_on_batch`, this will add a few milliseconds to each batch. Each comparison is ~3-4ms. Times the number of model inputs...

Is there a faster way to check this?

Reply:
Working on it. I'll do some test myself and try to figure it out.

Reply:
@fchollet can you give a hint of what input did you feed into `train_on_batch` ?